internal/infrastructure/kubernetes/service.go

```diff
@@ -116,3 +116,21 @@ func (i *Infra) createService(ctx context.Context, infra *ir.Infra) (*corev1.Ser
 
 	return expected, nil
 }
+
+// deleteService deletes the Envoy Service in the kube api server, if it exists.
+func (i *Infra) deleteService(ctx context.Context) error {
+	svc := new(corev1.Service)
+	key := types.NamespacedName{
+		Namespace: i.Namespace,
+		Name:      envoyServiceName,
+	}
+	err := i.Client.Get(ctx, key, svc)
```

Comment:
do you want to upate this too to look like `deleteDeployment` which executes ` i.Client.Delete` and checks `kerrors.IsNotFound(err)` to decide ret err ?